### PR TITLE
camerad: protect concurrent access to shared variables and avoid race conditions

### PR DESCRIPTION
**Pull request by deanlee**

### CODE REVIEW

File: selfdrive/camerad/cameras/camera_qcom.cc [hunk 1]
```diff
@@ -1732,7 +1734,9 @@ void actuator_move(CameraState *s, uint16_t target) {
   //LOGD("actuator move focus: %d", err);
 
   s->cur_step_pos = dest_step_pos;
+  pthread_mutex_lock(&s->frame_info_lock);
   s->cur_lens_pos = actuator_cfg_data.cfg.move.curr_lens_pos;
```
Comment (ZwX1616) [inline selfdrive/camerad/cameras/camera_qcom.cc:1738]: is this necessary?

Comment (deanlee) [reply selfdrive/camerad/cameras/camera_qcom.cc:1738]: it looks like that this value will be written in ops_thread(by calling do_autofocus) and read in the main thread.did I get that right?

Comment (ZwX1616) [reply selfdrive/camerad/cameras/camera_qcom.cc:1738]: the one constantly being written is called true_len_pos? I think the cur_lens_pos should be just a read-only number returned from the device.

File: selfdrive/camerad/main.cc [hunk 2]
```diff
@@ -1,6 +1,7 @@
 #include <stdio.h>
 #include <signal.h>
 #include <cassert>
+#include <mutex>
```
Comment (adeebshihadeh) [inline selfdrive/camerad/main.cc:4]: looks like this is only used in camera_qcom?

**Comment by adeebshihadeh** [comment]
Can you rebase, merging one of your other PRs caused a merge conflict?

**Comment by deanlee** [comment]
Fixed merge conflict

**Pull request merged**